shared/src/commonMain/kotlin/dev/johnoreilly/confetti/ConfettiRepository.kt

```diff
@@ -150,11 +150,15 @@ class ConfettiRepository : KoinComponent {
         }
     }
 
-    suspend fun sessionDetails(
+    fun sessionDetails(
```

Comment:
It feels like we don't have a common plan for how to implement these.  A similar issue happens with Room, for suspend functions vs Flow.

But especially here, it would be nice if we could hide away differences between CacheFirst, CandAndNetwork, NetworkFirst.

I'm unclear what happens when CacheAndNetwork, gets a valid cache result, but then fails on network.  Does it silently drop it, so the cache result stands? Or show an error?

Reply:
>  what happens when CacheAndNetwork, gets a valid cache result, but then fails on network

It emits 2 `ApolloResponse`:

* First one with `apolloResponse.exception == null`
* Seconf one with `apolloResponse.exception != null`

Reply:
What about the data?

Should we ignore that second one? assume it's just network is down?

Reply:
> What about the data?

* First one with `apolloResponse.data != null`
* Second one with `apolloResponse.data == null`

> Should we ignore that second one? assume it's just network is down?

I would ignore it yes. And rely on pull to refresh if the user really want latest data.

To me:

* initial load with `CacheAndNetwork` means: display something as soon as possible, update in the background
* pull to refresh means: I always want fresh network results (or an error else)

The more I think about it, the less I like `CacheAndNetwork` though because it updates the data without user action which might feel a bit weird.

Reply:
> The more I think about it, the less I like `CacheAndNetwork` though because it updates the data without user action which might feel a bit weird.

I like that, but it's a change for mobile.